sympy/polys/ring_series.py

```diff
@@ -1158,3 +1158,97 @@ def rs_compose_add(p1, p2):
     if dp:
         q = q*x**dp
     return q
+
+
+class RingSeriesBase(object):
```

Comment:
This is closer to `polys` than to `Basic`. The `series` in `RingSeries` is a `PolyElement` which can't be sympified. I am not sure they will work with `Basic`/`Expr`.